apps/designer/app/designer/features/sidebar-left/panels/pages/settings.tsx

```diff
@@ -7,26 +7,119 @@ import {
   useId,
   styled,
   Flex,
-  toast,
+  toastNonFieldErrors,
   InputErrorsTooltip,
 } from "@webstudio-is/design-system";
-import { useFetcher } from "@remix-run/react";
-import { ChevronDoubleLeftIcon } from "@webstudio-is/icons";
+import { useFetcher, type Fetcher } from "@remix-run/react";
+import { ChevronDoubleLeftIcon, TrashIcon } from "@webstudio-is/icons";
+import { utils as projectUtils } from "@webstudio-is/project";
 import type { ZodError } from "zod";
 import { Header } from "../../lib/header";
-import { useEffect, useRef, useState } from "react";
+import { useRef, useState, useCallback } from "react";
 import { type Page } from "@webstudio-is/project";
+import { usePages } from "~/designer/shared/nano-states";
+import { useDebounce, useUnmount } from "react-use";
+import { useOnFetchEnd, usePersistentFetcher } from "~/shared/fetcher";
 
 const Group = styled(Flex, {
   marginBottom: "$3",
   gap: "$2",
   defaultVariants: { direction: "column" },
 });
 
-export type CreatePageData =
-  | { ok: true; page: Page }
+type FetcherData<Payload> =
+  | ({ ok: true } & Payload)
   | { errors: string | ZodError["formErrors"] };
 
+const normalizeErrors = (
+  errors: string | ZodError["formErrors"]
+): ZodError["formErrors"] =>
+  typeof errors === "string"
+    ? { formErrors: [errors], fieldErrors: {} }
+    : errors;
+
+type EditablePage = Omit<Page, "treeId" | "id">;
+
+const FormFields = ({
+  disabled,
+  values,
+  onChange,
+  fetcher,
+}: {
+  disabled?: boolean;
+  values?: EditablePage;
+  onChange?: <FieldName extends keyof EditablePage>(
+    field: FieldName,
+    value: EditablePage[FieldName]
+  ) => void;
+  fetcher: Fetcher<FetcherData<unknown>>;
+}) => {
+  const fieldIds = {
+    name: useId(),
+    path: useId(),
+  } as const;
+
+  const fieldNames = useRef(Object.keys(fieldIds));
+
+  const [fieldErrors, setFieldErrors] = useState<
+    ZodError["formErrors"]["fieldErrors"]
+  >({});
+
+  useOnFetchEnd(fetcher, (data) => {
+    if ("errors" in data) {
+      const errors = normalizeErrors(data.errors);
+      toastNonFieldErrors(errors, fieldNames.current);
+      setFieldErrors(errors.fieldErrors);
+    }
+  });
+
+  const resetErrors = (fieldName: keyof typeof fieldIds) =>
+    setFieldErrors(({ [fieldName]: _, ...rest }) => rest);
+
+  return (
+    <>
+      <Group>
+        <Label htmlFor={fieldIds.name} size={2}>
```

Comment:
Label is based on __DEPRECATED__Text component, we need to update it to use Text component which doesn't have size prop any more, because we switched to semantic variants

Reply:
I can update it, but should this be a part of this PR?

Reply:
Not necessarily if you do it right next

Reply:
Will do a separate PR: https://github.com/webstudio-is/webstudio-designer/issues/381